### PR TITLE
#184347085 - email verification

### PR DESCRIPTION
**Pull request by teerenzo**
### What does this PR do?
User Verification Process 

### Description of Task to be completed?

this PR contains user verification via email by sending him/her verification link and whe he/she clickend on the link account  will verified. and also when user try to signIn without verifying his/her account he get message telling him that his account is not verified.

List any dependencies that are required for this change.

1. nodemailer
2. bcryptjs

### How should this be manually tested?

you can test this by running npm i

1. npm i
2. make sure you have all added variables in env
3. npm start
4. test using this endpoint http://localhost:3000/api/v1/users/signup or you can use http://localhost:3000/api-docs/

### Any background context you want to provide?

no other information

### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/184347085

### Questions:

# Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [X] My code is formatted correctly
- [x] My changes doesn't have any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by NG-Solennel — APPROVED** [review]

**Review by Marthatwesi — APPROVED** [review]

**Review by Ntarecedrick — APPROVED** [review]

File: src/database/migrations/20230131072323-create-user.js [hunk 1]
```diff
@@ -17,6 +17,9 @@ module.exports = {
       password: {
         type: Sequelize.STRING,
       },
+      isVerified: {
+        type: Sequelize.BOOLEAN,
```
Comment (denislohan) [inline src/database/migrations/20230131072323-create-user.js:21]: We need to have a default value: false

**Review by denislohan — CHANGES_REQUESTED** [review]

File: src/database/migrations/20230131072323-create-user.js (hunk 1 above)
Comment (teerenzo) [reply src/database/migrations/20230131072323-create-user.js:21]: done

**Review by NG-Solennel — APPROVED** [review]

**Review by Marthatwesi — APPROVED** [review]

File: src/utils/passport.config.js [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-        if (!isPin) {
-          done(null, false);
-        } else {
-          return done(null, userFound);
-        }
-      }
-    }
-  )
-);
+// passport.use(
```
Comment (Eli250) [inline src/utils/passport.config.js:6]: Why did you comment on these?

**Review by Eli250 — CHANGES_REQUESTED** [review]
You have  commented out all files  in `passport.config.js`, we need to understand why.

**Pull request closed without merging**